File: backend/api/app/models/cover.py
from __future__ import annotations

from pydantic import BaseModel


class CoverAsset(BaseModel):
    filename: str
    rel_path: str
    author: str
    book: str
    role: str
    variant: str
    url: str


class TypographyRegime(BaseModel):
    id: str
    name: str
    face: str
    alignment: str
    distinctive: str
    active: bool = False


class CoverAssetsResponse(BaseModel):
    assets: list[CoverAsset]
    total: int


class CoverRegimesResponse(BaseModel):
    regimes: list[TypographyRegime]
    active_regime: str | None

File: frontend/src/types/cover.ts
export interface CoverAsset {
  filename: string
  rel_path: string
  author: string
  book: string
  role: 'cover' | 'art' | string
  variant: string
  url: string
}

export interface TypographyRegime {
  id: string
  name: string
  face: string
  alignment: string
  distinctive: string
  active: boolean
}

export interface CoverAssetsResponse {
  assets: CoverAsset[]
  total: number
}

export interface CoverRegimesResponse {
  regimes: TypographyRegime[]
  active_regime: string | null
}
